kernel/kernel_drivers.go

```diff
@@ -148,41 +153,58 @@ func createModulesSubtree(kernelMount, kernelTree, kversion string, kmodsConts [
 		}
 	}
 
-	// Symbolic links to early mount kernel snap
-	earlyMntDir := filepath.Join(kernelMount, "modules", kversion)
+	// Symbolic links to current mount of the kernel snap
+	currentMntDir := filepath.Join(kMntPts.CurrentMntPt, "modules", kversion)
+	if err := createKernelModulesSymlinks(modsRoot, currentMntDir); err != nil {
+		return err
+	}
+
+	// If necessary, add modules from components and run depmod
+	if err := setupModsFromComp(kernelTree, kversion, comps); err != nil {
+		return err
+	}
+
+	// Change symlinks to target ones when needed
+	if kMntPts.CurrentMntPt != kMntPts.TargetMntPt {
+		targetMntDir := filepath.Join(kMntPts.TargetMntPt, "modules", kversion)
+		if err := createKernelModulesSymlinks(modsRoot, targetMntDir); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func createKernelModulesSymlinks(modsRoot, kMntPt string) error {
 	for _, d := range []string{"kernel", "vdso"} {
 		lname := filepath.Join(modsRoot, d)
-		to := filepath.Join(earlyMntDir, d)
+		to := filepath.Join(kMntPt, d)
+		// We might be re-creating, first remove
+		os.Remove(lname)
```

Comment:
do we need to do Remove before other Symlink calls ?

Reply:
This is the name of the link that we are creating, so I don't think there could be a collision, unless I am missing something here.